src/bokeh/models/glyphs.py

```diff
@@ -859,6 +860,27 @@ def __init__(self, *args, **kwargs) -> None:
         The color mapping step happens on the client.
     """)
 
+    def _validate_with_data_source(self, source: ColumnDataSource):
+        class_name = self.__class__.__name__
+
+        if not isinstance(self.color_mapper, WeightedStackColorMapper):
+            raise ValueError(f"{class_name}.color_mapper should be a WeightedStackColorMapper not a {self.color_mapper.__class__.__name__}")
+        n_colors = len(self.color_mapper.palette)
+        if n_colors < 1:
+            # This check should really be in ColorMapper.
+            raise ValueError(f"{class_name}.color_mapper.palette does not contain any colors")
```

Comment:
As a part of PR #13204 I'm working on constrained property types, so that it will be possible to express a simple constraint like this on the type level.